### ISSUE TITLE
"user error (Bitcode parsing of ffs-rust.bc failed" with Rust byte code

### ISSUE THREAD

**Issue by ttaubert** [GaloisInc/saw-script#84]
I took a small Rust library:

```
// test.rs
#[no_mangle]
pub extern fn ffs_ref(a: u32) -> u32 {
  return 0;
}
```

And compiled it to LLVM byte code with:

```
rustc --crate-type=staticlib --emit llvm-bc -o ffs-rust.bc ffs.rs
```

SAW unfortunately fails to parse the byte code emitted by Rust:

```
Loading module Cryptol
Loading file "ffs-rust.saw"
Extracting reference term
saw: user error (Bitcode parsing of ffs-rust.bc failed:
mzero
  from:
  	GLOBALVAR
  	MODULE_BLOCK)
```

I can call `ffs_ref()` just fine from a small C file compiled against the library built without `--emit`.

**Comment by ttaubert** [comment]
Here's the byte code: [ffs-rust.bc.zip](https://github.com/GaloisInc/saw-script/files/86347/ffs-rust.bc.zip)

**Comment by TomMD** [comment]
I'm not sure this issue is a duplicate, but just as a matter of feedback: it is known that the LLVM parser is not complete and does not yet handle code emitted by rust.  Tackling this issue is on our mental todo list.

**Comment by atomb** [comment]
I suspect that the issue here is the same as in #83, and that when we update the parser both will be fixed. However, it's nice to have examples of different failures, so that we're sure to address both.

**Comment by thenak** [comment]
I found this page trying to fix a similar error, but I'm using your tutorial example (ffs) on Windows. I tried creating the LLVM bitcode using different versions in clang windows and clang on linux.  Same error:

C:\galois\saw-0.1-dev-2015-06-09-Windows7Pro-32\doc\code>saw ffs_llvm.saw
  Loading module Cryptol
  Loading file "ffs_llvm.saw"
  Extracting reference term
  saw: user error (Bitcode parsing of ffs.bc failed:
  match failed)

It seems to fail at the very first instruction: 
l <- llvm_load_module "ffs.bc";

I'm running with the SAW version shown above and Cryptol 2.2.6.

**Comment by thenak** [comment]
^^ I was able to get clang 3.0 version working across environments, but not the 3.9.  Based on issue #83 I'm guessing it's just that this version of clang is not supported in the previous version of SAW.
Thanks

**Comment by kiniry** [comment]
As I wrote @thenak via email:

As you can see from the ticket, we are aware that we have to chase the undocumented tail of yet another LLVM release’s binary format.  Hopefully for now using a somewhat older version of clang will suffice.

**Comment by atomb** [comment]
This example now works with `rustc` version 1.7. It fails with the latest beta and nightly, but that should be fixed along with #122.

**Comment by atomb** [comment]
This now works with the latest LLVM bitcode parser.
